atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/cleaner/DbKvsNamespaceDeleterFactory.java

```diff
@@ -0,0 +1,116 @@
+            return createOracleNamespaceDeleter(dbKeyValueServiceConfig, runtimeConfig);
+        }
+        throw new SafeIllegalArgumentException(
+                "Dropping a namespace for the given DB type is not supported",
+                SafeArg.of("type", dbType),
+                SafeArg.of("supportedTypes", OracleDdlConfig.TYPE));
+    }
+
+    private NamespaceDeleter createOracleNamespaceDeleter(
+            DbKeyValueServiceConfig config, Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig) {
+        OracleDdlConfig ddlConfig = (OracleDdlConfig) config.ddl();
+        OracleConnectionConfig connectionConfig =
+                getConfigWithSingleConnection((OracleConnectionConfig) config.connection());
+
+        OracleTableNameGetter tableNameGetter = OracleTableNameGetterImpl.createDefault(ddlConfig);
+        OraclePrefixedTableNames prefixedTableNames = new OraclePrefixedTableNames(tableNameGetter);
+        TableValueStyleCache cache = new TableValueStyleCache();
+
+        // DDL tables require a compaction executor service. However, namespace deletion doesn't use compaction at
+        // all, so we never use the executor service.
+        ExecutorService executorService = MoreExecutors.newDirectExecutorService();
+        OracleDbTableFactory factory =
+                new OracleDbTableFactory(ddlConfig, tableNameGetter, prefixedTableNames, cache, executorService);
+        ConnectionSupplier connectionSupplier = createConnectionSupplier(config, runtimeConfig, connectionConfig);
+
+        OracleNamespaceDeleterParameters parameters = OracleNamespaceDeleterParameters.builder()
+                .userId(connectionConfig.getDbLogin())
+                .tablePrefix(ddlConfig.tablePrefix())
+                .overflowTablePrefix(ddlConfig.overflowTablePrefix())
+                .connectionSupplier(connectionSupplier)
+                .oracleDdlTableFactory(tableReference -> factory.createDdl(tableReference, connectionSupplier))
+                .tableNameGetter(tableNameGetter)
+                .build();
+
+        return new OracleNamespaceDeleter(parameters);
+    }
+
+    private ConnectionSupplier createConnectionSupplier(
+            DbKeyValueServiceConfig config,
+            Refreshable<Optional<KeyValueServiceRuntimeConfig>> runtimeConfig,
+            ConnectionConfig connectionConfig) {
+        ConnectionManager connectionManager = HikariClientPoolConnectionManagers.create(connectionConfig);
+        SqlConnectionSupplier sqlConnSupplier =
+                SqlConnectionSuppliers.createSimpleConnectionSupplier(connectionManager, config, runtimeConfig);
+        return new ConnectionSupplier(sqlConnSupplier);
+    }
+
+    private OracleConnectionConfig getConfigWithSingleConnection(OracleConnectionConfig config) {
+        return new OracleConnectionConfig.Builder()
+                .from(config)
+                .minConnections(0)
+                .maxConnections(1)
+                .build();
```

Comment:
There was the option to manage a shared pool e.g if there's a client that will delete multiple namespaces (i.e., internal proxy). However, given the rate of polling (1 hour), and the fact that there's only 1-2 namespaces deleted at a time (due to rate limiting), I've taken the simpler option of just opening and closing connections as needed.

Reply:
This is much simpler IMO